c_common/front_end_common_lib/src/recording.c

```diff
@@ -290,151 +289,142 @@ static inline bool _recording_write_memory(
             uint8_t *write_pointer =
-                g_recording_channels[channel].dma_current_write;
-            uint8_t *read_pointer = g_recording_channels[channel].current_read;
+                g_recording_channels[i].dma_current_write;
+            uint8_t *read_pointer = g_recording_channels[i].current_read;
             buffered_operations last_buffer_operation =
-                g_recording_channels[channel].last_buffer_operation;
+                g_recording_channels[i].last_buffer_operation;
 
             if (read_pointer < write_pointer) {
-                _create_buffer_message(
-                    data_ptr, n_requests, channel, read_pointer,
+                create_buffer_message(
+                    data_ptr, n_requests, i, read_pointer,
                     write_pointer - read_pointer);
                 n_requests++;
             } else if ((write_pointer < read_pointer) ||
                     (write_pointer == read_pointer &&
                      last_buffer_operation == BUFFER_OPERATION_WRITE)) {
-                _create_buffer_message(
-                    data_ptr, n_requests, channel, read_pointer,
+                create_buffer_message(
+                    data_ptr, n_requests, i, read_pointer,
                     end_of_buffer_region - read_pointer);
                 n_requests++;
 
-                _create_buffer_message(
-                    data_ptr, n_requests, channel, buffer_region,
+                create_buffer_message(
+                    data_ptr, n_requests, i, buffer_region,
                     write_pointer - buffer_region);
                 n_requests++;
             } else {
-
                 // something somewhere went terribly wrong this should never
                 // happen
                 log_error(
                     "Unknown channel state - channel: %d, start pointer: %d,"
                     " end pointer: %d, read_pointer: %d, write_pointer: %d,"
-                    " last operation==READ: %d", channel, buffer_region,
+                    " last operation==READ: %d", i, buffer_region,
                     end_of_buffer_region, read_pointer, write_pointer,
                     last_buffer_operation == BUFFER_OPERATION_READ);
             }
         }
     }
 
     if (n_requests > 0) {
-
         // eieio command packet with command ID 8
         req_hdr->eieio_header_command = 0x4008;
         req_hdr->chip_id = spin1_get_chip_id();
         data_ptr[0].processor_and_request =
             (spin1_get_core_id() << 3) | n_requests;
         data_ptr[0].sequence = sequence_number;
         log_debug("Sending request with sequence %d", sequence_number);
-        msg_size += (n_requests * sizeof(read_request_packet_data));
+        msg_size += n_requests * sizeof(read_request_packet_data);
         msg.length = msg_size;
 
         spin1_send_sdp_msg(&msg, 1);
     }
 }
 
-static void _buffering_in_handler(uint mailbox, uint port) {
+static NOINLINE void buffering_in_handler(
```

Comment:
Again, check why not inlined...

Reply:
This one is used as a callback with `simulation_sdp_callback_on`.